resources/containers/field-modal.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from "react";
+import Modal from "react-modal";
+import styles from "./styles/field-modal.module.css";
+
+interface FieldModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onSave: (data: any) => void;
+  fieldType: string;
+}
+
+const FieldModal: React.FC<FieldModalProps> = ({
+  isOpen,
+  onClose,
+  onSave,
+  fieldType,
+}) => {
+  const [selectedOptions, setSelectedOptions] = useState<string[]>([]);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const options = ["number", "text"];
+
+  useEffect(() => {
+    setSelectedOptions([]);
+  }, [isOpen]);
+
+  const handleSave = () => {
+    if (fieldType === "number") {
+      onSave({ maxValue: parseInt(selectedOptions[0] || "0") });
+    } else if (fieldType === "multiselect") {
+      onSave({ options: selectedOptions });
+    }
+    onClose();
+  };
+
+  const handleOptionSelect = (option: string) => {
+    if (!selectedOptions.includes(option)) {
+      setSelectedOptions([...selectedOptions, option]);
+    }
+    setIsDropdownOpen(false);
+  };
+
+  const handleRemoveOption = (optionToRemove: string) => {
+    setSelectedOptions(
+      selectedOptions.filter((option) => option !== optionToRemove),
+    );
+  };
+
+  return (
+    <Modal
```

Comment:
The mmodal is a resource, even if is just to rehuse the style we add to the `react-modal`